src/View/SSViewer_Scope.php

```diff
@@ -339,6 +344,9 @@ public function next()
     public function __call($name, $arguments)
     {
         $on = $this->itemIterator ? $this->itemIterator->current() : $this->item;
+        if (is_scalar($on) || is_null($on)) {
+            $on = $this->convertPrimitiveToDBField($on);
+        }
```

Comment:
This is possibly me wanting "correctness" so if you have a good reason against doing this I won't push it - but I think the call to `convertPrimitiveToDBField()` should be done in `getItem()` instead, and make the following change everywhere that it can be made in this class and subclasses:

```diff
- $on = $this->itemIterator ? $this->itemIterator->current() : $this->item;
+ $on = $this->getItem();
```

That way scalars will always be treated the same way everywhere in scope, instead of only being treated this way when attempting to call methods on them from the template.

Reply:
Done

Reply:
There's an instance of `$this->itemIterator ? $this->itemIterator->current() : $this->item` in `SSViewer_DataPresenter` as well which is a subclass of `SSViewer_Scope`

Reply:
Still not done?

Reply:
Done